### PR TITLE
[ENH] Switch default and filesystem name envs

### PR DESCRIPTION
**Pull request by viniciusdc**
## Changes introduced in this PR:

- Update `filesystem` and `default` environment names to `nebari-git` and `global` respectfully 
 
related to #1349 

## Types of changes

What types of changes does your PR introduce?

_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [x] Breaking change (fix or feature that would cause existing features to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

### Requires testing

- [x] Yes
- [ ] No

### In case you checked yes, did you write tests?

- [ ] Yes
- [x] No

## Documentation

Does your contribution include breaking changes or deprecations?
If so have you updated the documentation?

- [ ] Yes, docstrings
- [ ] Yes, main documentation
- [ ] Yes, deprecation notices

## Further comments (optional)
This will require a full deployment of Qhub to validate all services run as excepted.

### CODE REVIEW

**Review by iameskild — DISMISSED** [review]
This looks good to me! I know we discussed this as a team but I still have some reservations about `nebari-git` as a namespace name. However given that this was a team decision, I'm fine with it.

As I was thinking this through a few questions came to mind. 
- What happens when we upgrade from `0.4.2` to `0.4.3`, does the `filesystem` and `default` namespaces just linger around? 
  - My assumption is that they will stick around but might not be useable.
- And if so, does this cause issues for new environments?
- And do the users need to recreate their old envs that were previously under the `filesystem` namespace? 
  - I assume when they redeploy, the envs in the config file will be automatically recreated under the `nebari-git` namespace but if there were other envs created using the conda-store UI, I don't think those will be.

**Comment by viniciusdc** [comment]
I think I am missing something 
image

**Comment by viniciusdc** [comment]
I am out of ideas now... @costrouc may you have a look if I am missing something? Even with a fresh install, the old namespaces were created. 

I've also tried using 
```
c.CondaStore.default_namespace = "global"
c.CondaStore.filesystem_namespace = "nebari-git"
```

**Comment by trallard** [comment]
@costrouc  this is waiting for your input ☝🏽

**Comment by trallard** [comment]
Blocked by [#conda](https://github.com/Quansight/qhub/pull/1385)

**Comment by iameskild** [comment]
@viniciusdc I opened this [PR in nebari-docker-images repo](https://github.com/nebari-dev/nebari-docker-images/pull/21) given that we recently decoupled the images. I will try to test this on quansight-beta 👍

**Comment by viniciusdc** [comment]
> @viniciusdc I opened this [PR in nebari-docker-images repo](https://github.com/nebari-dev/nebari-docker-images/pull/21) given that we recently decoupled the images. I will try to test this as well, given the concerns I raised above +1

Thanks, @iameskild. I will then focus on updating this one. Also, I forgot to answer your points above, apologies, I do believe the old environments will still be in there, but some of the services might not show them. The users will definitely need to recreate the environments in the new namespaces as I don't think conda-store will treat those changes as a proper "renaming."

**Comment by viniciusdc** [comment]
Ok... so this will need to be updated:
- `default` -> `global` -> `nebari-system`

@costrouc Is there a way to hide this environment from the user?

**Comment by costrouc** [comment]
@viniciusdc yes we just drop permissions for users with this given namespace and they will no longer see it.

File: tests_deployment/test_jupyterhub_ssh.py [hunk 1]
```diff
@@ -94,7 +94,10 @@ def test_exact_jupyterhub_ssh(paramiko_object):
         ("whoami", constants.KEYCLOAK_USERNAME),
         ("pwd", f"/home/{constants.KEYCLOAK_USERNAME}"),
         ("echo $HOME", f"/home/{constants.KEYCLOAK_USERNAME}"),
-        ("conda activate default && echo $CONDA_PREFIX", "/opt/conda/envs/default"),
```
Comment (costrouc) [inline tests_deployment/test_jupyterhub_ssh.py:97]: I don't think this should be changed.

File: qhub/template/stages/07-kubernetes-services/modules/kubernetes/services/jupyterhub/main.tf [hunk 2]
```diff
@@ -50,7 +50,7 @@ resource "helm_release" "jupyterhub" {
               kind      = "configmap"
             }
 
-            "/opt/conda/envs/default/share/jupyter/lab/settings" = {
```
Comment (costrouc) [inline qhub/template/stages/07-kubernetes-services/modules/kubernetes/services/jupyterhub/main.tf:53]: I don't think this should be changed. This is the name of the conda env from the docker image.

**Review by costrouc — DISMISSED** [review]

**Comment by viniciusdc** [comment]
Thanks @HarshCasper I will have a look and test this as well!!! :smile:

**Comment by viniciusdc** [comment]
@HarshCasper based on what we discussed today, we will need to make the `nebari-git` a customizable name in the config -- we can use the input variables and the ztjb custom config to pass through the values.

Regarding the name for `global,` we need to make sure this is just used in the installation part of jupyterhub/jupyterlab -- I am inclined to say that's the only case. Still, we can verify by checking wich env is used in the user pod terminal...

To hide an environment, we can change the permission part of global to admin only -- this should hide for the others...

**Review by iameskild — COMMENTED** [review]
I tested this on an existing deployment and it was successful. I was able to create an env in the `global` namespace.

As a team, we should also decide how we want to handle this for existing users as the existing namespaces will be sticking around. Some possible options include:
- an `initContainer` that will move / clean the old namespaces - moderately challenging
- provide instructions on how to delete the old namespaces - simply but a bit hostile
  - this might also include rebuilding the envs that were once in the `filesystem` namespace

cc @viniciusdc

**Comment by trallard** [comment]
TODO: add troubleshooting section for folks upgrading to `0.4.5` and reference section on envs

**Review by iameskild — APPROVED** [review]
Thanks a lot for seeing this through to the end @viniciusdc :)

**Pull request merged**